### DISCUSSION TITLE
HowTo use sort query parameter when using rest_framework_json_api.filters.OrderingFilter

### DISCUSSION THREAD

**Discussion by n2ygk** [jokiefer/drf-spectecular-json-api#28] [Q&A] [answered]
I've been playing with https://openapi-generator.tech (how-to [described here](https://github.com/columbia-it/django-jsonapi-training/blob/openapi-generator/docs/browser_client.md)) to automatically generate all the necessary models and services for an Angular single page app  based on the drf-spectacular-json-api generated OAS schema doc. `sort` is missing from the generated query parameters ([example](https://github.com/columbia-it/django-jsonapi-training/blob/openapi-generator/frontend/oas-angular-app/src/app/core/api/v1/api/courses.service.ts#L71-L130)). The sort filter is configured at https://github.com/columbia-it/django-jsonapi-training/blob/openapi-generator/training/settings.py#L195-L200 in the `DEFAULT_FILTER_BACKENDS`.

**Comment by jokiefer** [comment, marked as answer]
The support for sort description is done by now.
If you define the `ordering_fields` in your class based view for example, the sort parameter shout be injected like:

```json
{
    "name": "sort",
    "required": false,
    "in": "query",
    "description": "[list of fields to sort by](https://jsonapi.org/format/#fetching-sorting)",
    "schema": {
        "type": "array",
        "items": {
            "type": "string",
            "enum": [
                "id",
                "-id",
                "username",
                "-username",
                "firstName",
                "-firstName",
                "lastName",
                "-lastName",
                "email",
                "-email",
                "isActive",
                "-isActive",
                "dateJoined",
                "-dateJoined"
            ]
        }
    },
    "explode": false
}
```

Class based View example:

```python
class WebMapServiceViewSet(
    ModelViewSet
):
    ....
    search_fields = ("id", "title", "abstract", "keywords__keyword")
    ordering_fields = ["id", "title", "abstract", "hits", "date_stamp"]
```

Maybe you did miss this definitions on your views?



Little hint: It seems you are generating the client models statically based on the openapi schema. 
Maybe it is also interesting for you to dig deeper in my own project, where i generate the client dynamically from the schema so i do not need to reproduce the models on the client side.

Here is a [CreateGuesser](https://github.com/mrmap-community/mrmap/blob/master/frontend/src/jsonapi/components/CreateGuesser.tsx) component, based on [react admin](https://marmelab.com/react-admin/documentation.html), where i get the different inputs for the form from the openapi definitions. There is also an EditGuesser, ListGuesser and so on. For now it is part of the main project. But in the future it could be a separated package.

If you are interested in it, feel free to start discussions on the mrmap project.

**Reply by n2ygk** [reply]
Ah ok. I believe I did not specify ordering_fields since all fields are available for ordering by default when the OrderingFilter is present. I can certainly add them. See https://www.django-rest-framework.org/api-guide/filtering/

Will look at your dynamic stuff as well. Very cool.
